lib/alerts/survey_response_alert_manager.rb

```diff
@@ -9,21 +9,33 @@ def create_alerts
     @answers.each do |answer|
       next unless answer_meets_alert_conditions?(answer)
       next if alert_already_exists?(answer)
-      details = {
-        question: answer.question.question_text,
-        answer: answer.answer_text,
-        followup: answer.follow_up_answer_text
-      }
       alert = Alert.create(type: 'SurveyResponseAlert',
                            message: alert_message(answer),
                            user_id: answer.user.id,
                            subject_id: answer.question.id,
-                           details: details,
+                           details: details(answer),
                            target_user_id: alert_recipient&.id)
       alert.email_target_user
     end
   end
 
+  def details(answer)
+    {
+      question: answer.question.question_text,
+      answer: answer.answer_text,
+      followup: answer.follow_up_answer_text,
+      source: self.class.source(answer)
+    }
+  end
+
+  def self.source(answer)
```

Comment:
Is this just a class method so that it can be called from the preview?

Reply:
Actually, no. Method was too long and rubocop raised an offense. Si I made a method of detail;

Reply:
Doing this as a class method is an unusual way to handle it, unless you need to be able to call it from somewhere else without having an instance already. It would be more idiomatic Ruby to use a private instance method  here.